### ISSUE TITLE
After upgrade to 7.x - it is impossible send one message with multiple recipients

### ISSUE THREAD

**Issue by vgrebenschikov** [anymail/django-anymail#179]
* django-anymail[sendgrid]==7.0.0
* EMAIL_BACKEND = 'anymail.backends.sendgrid.EmailBackend'
* Django==1.11.20, requests==2.21.0, python 3.7.4
* AssertionError at sengrid.py:137 in build_merge_data
```
assert len(personalization["to"]) == 1
```

Looks like problem was introduced in [d2d568b6d3](https://github.com/anymail/django-anymail/commit/d2d568b6d33acf0a0893dd107d96af02f5993946)
(@medmunds - can you advice was it intentional?)

before - assert on multiple personalizations:
```
assert len(self.data["personalizations"]) == 1
```
after - assert on multiple recipients in one mail:
```
assert len(personalization["to"]) == 1
```
Which completely broke my case when multiple recipients are specified in to_addresses (see below)

Stacktrace:
> File ".../django/core/mail/message.py", line 348, in send
>      return self.get_connection(fail_silently).send_messages([self])
> File ".../anymail/backends/base.py", line 95, in send_messages
>      sent = self._send(message)
> File ".../anymail/backends/base_requests.py", line 60, in _send
>      return super(AnymailRequestsBackend, self)._send(message)
> File ".../anymail/backends/base.py", line 125, in _send
>      response = self.post_to_esp(payload, message)
> File ".../anymail/backends/base_requests.py", line 71, in post_to_esp
>      params = payload.get_request_params(self.api_url)
> File ".../anymail/backends/base_requests.py", line 166, in get_request_params
>      data=self.serialize_data(),
> File ".../anymail/backends/sendgrid.py", line 104, in serialize_data
>      self.build_merge_data()
> File ".../anymail/backends/sendgrid.py", line 137, in build_merge_data
>      assert len(personalization["to"]) == 1
> AssertionError

```
from django.core.mail import EmailMultiAlternatives

def send_mail(
    template_id, subject='', body='', body_html='', from_address=None, to_addresses=None,
    context=None,
):
    msg = EmailMultiAlternatives(subject, body, from_address, to_addresses)

    if template_id:
        msg.merge_global_data = context
        msg.template_id = template_id

    if body_html:
        msg.attach_alternative(body_html, 'text/html')

    try:
        msg.send()
    except AnymailError:
        logger.exception("Failed to send message with Anymail")
```

**Comment by medmunds** [comment]
Can I confirm that your use case is:
* You are using SendGrid templates
* You are *only* setting `merge_global_data` to fill in template variables (not `merge_data`), so every recipient gets the same templated message
* And you want *every* recipient on the `to` list to be able to see the email address of *every other recipient?* (You want to send a single email with all to addresses visible in it.)

It does look like there was a case missed by the refactoring in v6.0. Sorry about that. We should change the [assertion on line 137](https://github.com/anymail/django-anymail/blob/v7.0/anymail/backends/sendgrid.py#L137) to allow using templates with only `merge_global_data` and multiple recipients. (The assertion should really be `assert len(personalization["to"]) == 1 or len(self.merge_data) == 0`, and we'll want to add a test for templates with only `merge_global_data` and multiple `to` emails.)

If you *didn't* want every recipient to see the other recipients' email addresses, then you also need to set `message.merge_data = {}` to trigger batch send mode, with a separate, private email to each of the `to` addresses—see the Caution box in Anymail's [batch send docs](https://anymail.readthedocs.io/en/stable/sending/templates/#batch-sending-with-merge-data). (And setting `merge_data` would also be a workaround until this issue gets fixed, unless it's critical that all recipients see each other's addresses.)

**Comment by medmunds** [comment]
Fix released in Released in Anymail v7.1
